controllers/kubevirtmachine_controller.go

```diff
@@ -203,36 +210,53 @@ func (r *KubevirtMachineReconciler) reconcileNormal(ctx *context.MachineContext)
 		return ctrl.Result{}, errors.Wrap(err, "failed to fetch ssh keys for cluster nodes")
 	}
 
-	if err := r.reconcileKubevirtBootstrapSecret(ctx, clusterNodeSshKeys); err != nil {
-		ctx.Logger.Info("Waiting for the Bootstrap provider controller to set bootstrap data")
+	infraClusterClient, infraClusterNamespace, err := r.InfraCluster.GenerateInfraClusterClient(ctx.ClusterContext())
+	if err != nil {
+		ctx.Logger.Error(err, "Infra cluster client is NOT available.")
+	}
+	if infraClusterClient == nil {
+		ctx.Logger.Info("Waiting for infra cluster client...")
+		return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	ctx.Logger.Info("Reconciling bootstrap secret...")
+	if err := r.reconcileKubevirtBootstrapSecret(ctx, infraClusterClient, infraClusterNamespace, clusterNodeSshKeys); err != nil {
+		ctx.Logger.Info("Waiting for the Bootstrap provider controller to set bootstrap data...")
 		conditions.MarkFalse(ctx.KubevirtMachine, infrav1.VMProvisionedCondition, infrav1.WaitingForBootstrapDataReason, clusterv1.ConditionSeverityInfo, "")
 		return ctrl.Result{}, nil
 	}
 
+	ctx.Logger.Info("Checking if VM already exists...")
 	// Create a helper for managing the KubeVirt VM hosting the machine.
-	externalMachine, err := kubevirthandler.NewMachine(ctx, r.Client)
+	externalMachine, err := kubevirthandler.NewMachine(ctx, infraClusterClient, infraClusterNamespace)
```

Comment:
Right now, the namespace for the external KubeVirt VMs is inferred by introspecting the kubeconfig data for the external config. I'm unsure about that.

I think the issue we're trying to solve here is how to express the location of the underlying KubeVirt VM from the KubeVirtMachine's namespace.

One way to do this is to modify our KubeVirtMachine API to use a VMI template instead of a VMI Spec.

so, This 
```
// KubevirtMachineSpec defines the desired state of KubevirtMachine.                                                           
type KubevirtMachineSpec struct {                                                                                              
        VMSpec kubevirtv1.VirtualMachineInstanceSpec `json:"vmSpec,omitempty"`                                                 
                                                                                                                               
```

Could change to this...

```
// KubevirtMachineSpec defines the desired state of KubevirtMachine.                                                           
type KubevirtMachineSpec struct {                                                                                              
        VMITemplate kubevirtv1.VirtualMachineInstanceTemplateSpec `json:"vmiTemplate,omitempty"`                                                                                                                                                                                                               

```

The advantage here is that the VMITemplate object has a ObjectMetadata field, where we can specify labels,annotations and even the `namespace` of the VMI independently of the KubeVirtMachine.